### ISSUE TITLE
clang-format regression around class declarations with attributes

### ISSUE THREAD

**Issue by kadircet** [llvm/llvm-project#127470]
desired formatting:
```cpp
#define ABSL_ATTRIBUTE_TRIVIAL_ABI
#define ABSL_NULLABILITY_COMPATIBLE

class ABSL_ATTRIBUTE_TRIVIAL_ABI ABSL_NULLABILITY_COMPATIBLE
    ArenaSafeUniquePtr{};
```

actual formatting:
```cpp
#define ABSL_ATTRIBUTE_TRIVIAL_ABI
#define ABSL_NULLABILITY_COMPATIBLE

class ABSL_ATTRIBUTE_TRIVIAL_ABI
    ABSL_NULLABILITY_COMPATIBLE ArenaSafeUniquePtr {};
```

bisection points to https://github.com/llvm/llvm-project/pull/124891. it's similar to https://github.com/llvm/llvm-project/issues/124574 as we were putting class-name to its own line and now we're mixing it with attributes. I guess this is due to special casing `ClassHeadName` despite having a heuristic with false positives that determines it (e.g. in this case we're falsely treating `ABSL_ATTRIBUTE_TRIVIAL_ABI` as `ClassHeadName`).

**Comment by owenca** [comment]
> desired formatting:
> 
> #define ABSL_ATTRIBUTE_TRIVIAL_ABI
> #define ABSL_NULLABILITY_COMPATIBLE
> 
> class ABSL_ATTRIBUTE_TRIVIAL_ABI ABSL_NULLABILITY_COMPATIBLE
>     ArenaSafeUniquePtr{};

Expected format is:
```c++
class ABSL_ATTRIBUTE_TRIVIAL_ABI ABSL_NULLABILITY_COMPATIBLE
    ArenaSafeUniquePtr {};
```
Notice that there's a space before the `{`.

> I guess this is due to special casing `ClassHeadName` despite having a heuristic with false positives that determines it (e.g. in this case we're falsely treating `ABSL_ATTRIBUTE_TRIVIAL_ABI` as `ClassHeadName`).

Actually, before #124891:
```
AnnotatedTokens(L=0, P=0, T=6, C=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=class L=5 PPK=2 FakeLParens= FakeRParens=0 II=0x14b80d7f8 Text='class'
 M=0 C=1 T=Unknown S=1 F=0 B=0 BK=0 P=5020 Name=identifier L=32 PPK=2 FakeLParens= FakeRParens=0 II=0x14b811f08 Text='ABSL_ATTRIBUTE_TRIVIAL_ABI'
 M=0 C=1 T=StartOfName S=1 F=0 B=0 BK=0 P=220 Name=identifier L=60 PPK=2 FakeLParens= FakeRParens=0 II=0x14b811f50 Text='ABSL_NULLABILITY_COMPATIBLE'
 M=0 C=1 T=StartOfName S=1 F=0 B=0 BK=0 P=130 Name=identifier L=79 PPK=2 FakeLParens= FakeRParens=0 II=0x14b811f90 Text='ArenaSafeUniquePtr'
 M=0 C=0 T=ClassLBrace S=1 F=0 B=0 BK=1 P=23 Name=l_brace L=81 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='{'
```
After that PR:
```
AnnotatedTokens(L=0, P=0, T=6, C=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=class L=5 PPK=2 FakeLParens= FakeRParens=0 II=0x13880a7f8 Text='class'
 M=0 C=1 T=Unknown S=1 F=0 B=0 BK=0 P=5020 Name=identifier L=32 PPK=2 FakeLParens= FakeRParens=0 II=0x13881df08 Text='ABSL_ATTRIBUTE_TRIVIAL_ABI'
 M=0 C=1 T=StartOfName S=1 F=0 B=0 BK=0 P=220 Name=identifier L=60 PPK=2 FakeLParens= FakeRParens=0 II=0x13881df50 Text='ABSL_NULLABILITY_COMPATIBLE'
 M=0 C=0 T=ClassHeadName S=1 F=0 B=0 BK=0 P=23 Name=identifier L=79 PPK=2 FakeLParens= FakeRParens=0 II=0x13881df90 Text='ArenaSafeUniquePtr'
 M=0 C=0 T=ClassLBrace S=1 F=0 B=0 BK=1 P=23 Name=l_brace L=81 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='{'
```
So it got the `ClassHeadName` right, but didn't set `C` (Can break before).

**Comment by kadircet** [comment]
ah sorry for the confusion, not sure how I misread the annotations. thanks a lot for the quick fix!
